commands/summary.go

```diff
@@ -289,6 +289,9 @@ func addPprofData(ctx context.Context, zipW *zip.Writer, skipServer bool) {
 func (cmd *summaryCommand) makeArchive(ctx context.Context) (err error) {
 	var f *os.File
 
+	if cmd.Filename == "" {
+		cmd.Filename = filename
+	}
```

Comment:
I think it should be in `RunWithContext` before calling `makeArchive` because it's more related to input validation then to actual make archive logic.